nixos.nix

```diff
@@ -124,14 +154,51 @@ in
             directories = cfg.${persistentStoragePath}.directories;
             files = unique (map dirOf cfg.${persistentStoragePath}.files);
           in
-          nameValuePair
-            "createDirsIn-${replaceStrings [ "/" "." " " ] [ "-" "" "" ] persistentStoragePath}"
-            (noDepEntry (concatMapStrings
-              (mkDirWithPerms persistentStoragePath)
-              (directories ++ files)
-            ));
+          {
+            "createDirsIn-${replaceStrings [ "/" "." " " ] [ "-" "" "" ] persistentStoragePath}" =
+              noDepEntry (concatMapStrings
+                (mkDirWithPerms persistentStoragePath)
+                (directories ++ files));
+          };
+
+        dirCreationScripts = foldl' recursiveUpdate { } (map mkDirCreationScriptForPath persistentStoragePaths);
+
+        bindOrLinkFileScript = persistentStoragePath: file:
+          let
+            targetFile = escapeShellArg (concatPaths [ persistentStoragePath file ]);
+            mountPoint = escapeShellArg file;
+          in
+          {
+            "bindOrLink-${sanitizeName targetFile}" = {
+              deps = [ "createDirsIn-${replaceStrings [ "/" "." " " ] [ "-" "" "" ] persistentStoragePath}" ];
+              text = ''
+                if [[ -L ${mountPoint} && $(readlink -f ${mountPoint}) == ${targetFile} ]]; then
+                    echo "${mountPoint} already links to ${targetFile}, ignoring"
+                    exit 0
+                elif mount | grep -F ${mountPoint}' ' >/dev/null && ! mount | grep -F ${mountPoint}/ >/dev/null; then
+                    echo "mount already exists at ${mountPoint}, ignoring"
+                    exit 0
```

Comment:
@colemickens Thanks for testing! I think this is the culprit. It looks like a copy-paste-fail.

Reply:
@colemickens Can you try it now?